examples/cpp/dds/CustomPayloadPoolExample/CustomPayloadPoolDataPublisher.cpp

```diff
@@ -0,0 +1,210 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file CustomPayloadPoolDataPublisher.cpp
+ *
+ */
+
+#include "CustomPayloadPoolDataPublisher.h"
+#include <fastrtps/attributes/ParticipantAttributes.h>
+#include <fastrtps/attributes/PublisherAttributes.h>
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/publisher/Publisher.hpp>
+#include <fastdds/dds/publisher/qos/PublisherQos.hpp>
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/publisher/qos/DataWriterQos.hpp>
+#include <rtps/history/TopicPayloadPoolRegistry.hpp>
+
+#include <thread>
+
+using namespace eprosima::fastdds::dds;
+
+CustomPayloadPoolDataPublisher::CustomPayloadPoolDataPublisher(
+        std::shared_ptr<CustomPayloadPool> payload_pool)
+    : participant_(nullptr)
+    , publisher_(nullptr)
+    , topic_(nullptr)
+    , writer_(nullptr)
+    , type_(new CustomPayloadPoolDataPubSubType())
+    , payload_pool_(payload_pool)
+{
+    matched_ = 0;
+    first_connected_ = false;
+}
+
+bool CustomPayloadPoolDataPublisher::init()
+{
+    hello_.index(0);
+    hello_.message("CustomPayloadPool");
+    DomainParticipantQos pqos = PARTICIPANT_QOS_DEFAULT;
+    pqos.name("Participant_pub");
+    auto factory = DomainParticipantFactory::get_instance();
+
+    participant_ = factory->create_participant(0, pqos);
+
+    if (participant_ == nullptr)
+    {
+        return false;
+    }
+
+    //REGISTER THE TYPE
```

Comment:
Absolute NIT: I don't like this style of comments at all, all in caps and with a space I mean. For sections of code I normally do. If considered, please apply everywhere

```suggestion
    /* Register the type */
```